updated mypy, dev requirements and contributing.md

Goldziher opened the pull request:
This PR does the following:

1. update the pinned version of mypy to 0.782 (latest), which was my original intention (it makes my life miserable in my current project because Im forced to use mypy 0.070)
2. updates the dev dependencies
3. updates the configs and contributing.md
4. resolves all issues due to isort upgrade and missing flake8 configs.

sobolevn reviewed (changes requested):
Thanks a lot! Can you please fix the CI?

Goldziher commented:
@sobolevn I fixed CI and updated the configs. Apparently the flake8 config was missing entirely. I can fix all the import errors it finds - but this will pollute the diff (its already quite large due to the isort upgrade), so I am going to separate this to a followup PR.

sobolevn commented on rest_framework-stubs/serializers.pyi:
```diff
@@ -1,86 +1,29 @@
     Union,
 )
 
-from rest_framework.fields import (
```
We need this re-export. Because `mypy` treats this with a special care.

sobolevn commented on rest_framework-stubs/serializers.pyi:
```diff
@@ -1,86 +1,29 @@
-    Throttled as Throttled,
-    UnsupportedMediaType as UnsupportedMediaType,
-)
-from rest_framework.relations import (
```
Basically, we need all deleted re-exports.

Goldziher replied:
done

Goldziher replied:
done

sobolevn commented on rest_framework-stubs/test.pyi:
```diff
@@ -1,9 +1,11 @@
 from django.db.models import Model
 from django.http import HttpRequest, HttpResponse
 from django.test import testcases
-from django.test.client import Client as DjangoClient, ClientHandler, RequestFactory as DjangoRequestFactory
```
The same here.

sobolevn commented on setup.py:
```diff
@@ -19,7 +19,7 @@ def find_stub_files(name):
     readme = f.read()
 
-dependencies = ["mypy>=0.750,<0.780", "django-stubs>=1.3.0", "typing-extensions"]
+dependencies = ["mypy>=0.750", "django-stubs>=1.3.0", "typing-extensions"]
```
Let's bump `mypy` to `0.782` and `django-stubs` to `1.6`

Goldziher replied:
done

sobolevn commented on rest_framework-stubs/authentication.pyi:
```diff
@@ -1,7 +1,6 @@
 from typing import Any, Optional, Tuple, Type
 
 from django.db.models import Model
-from django.contrib.auth import authenticate as authenticate
```
The same here.

sobolevn commented on rest_framework-stubs/fields.pyi:
```diff
@@ -7,29 +7,28 @@ from typing import (
 from django.core.exceptions import ValidationError
 from django.db import models
 from django.db.models import Model
-from django.forms import FilePathField as DjangoFilePathField, ImageField as DjangoImageField
```
The same here.

sobolevn commented on rest_framework-stubs/schemas/__init__.pyi:
```diff
@@ -3,7 +3,9 @@ from typing import Any, Callable, Optional, Sequence, Type
 from rest_framework.renderers import BaseRenderer
 
 from .generators import SchemaGenerator as SchemaGenerator
-from .inspectors import AutoSchema as AutoSchema, DefaultSchema as DefaultSchema, ManualSchema as ManualSchema
```
The same here.

sobolevn reviewed (changes requested):
Thanks a lot! Really great work!

There's one major point we need to fix: re-exports.
Because, `flake8` reports them as unusued imports, but we need them for `mypy`.

Goldziher commented:
Thanks for the review! I addressed all comments

sobolevn commented:
Thanks a lot for your work!
